kDriveCore/Data/Cache/DriveFileManager/DriveFileManager+AvailableOffline.swift

```diff
@@ -35,81 +35,44 @@ public extension DriveFileManager {
         let offlineFiles = getAvailableOfflineFiles()
         guard !offlineFiles.isEmpty else { return }
 
-        let updatedFileResult = try await getUpdatedAvailableOffline()
+        let batchSize = 500
+        for i in stride(from: 0, through: offlineFiles.count - 1, by: batchSize) {
+            let batchFiles = Array(offlineFiles.dropFirst(i).prefix(batchSize))
+            let batchActivities = try await apiFetcher.filesLastActivities(files: batchFiles, drive: drive)
 
-        try database.writeTransaction { writableRealm in
-            for updatedFile in updatedFileResult.updatedFiles {
-                updateFile(updatedFile: updatedFile, writableRealm: writableRealm)
-            }
-
-            for deletedFileUid in updatedFileResult.deletedFileUids {
-                deleteOfflineFile(uid: deletedFileUid, writableRealm: writableRealm)
+            var updatedFiles = [File]()
+            try database.writeTransaction { writableRealm in
+                for activity in batchActivities {
+                    if let file = activity.file,
+                       [FileActivityType.fileUpdate, FileActivityType.fileRename].contains(activity.lastAction) {
+                        updateFile(updatedFile: file, lastActionAt: activity.lastActionAt, writableRealm: writableRealm)
+                        updatedFiles.append(file)
```

Comment:
Once the files are attached to a realm, can we freeze them, or am I being overcautious ?

Reply:
Good catch. We could be on other thread for each batch

Reply:
It's okay to not freeze it since we only use it on the same thread ln:60. Between each array updatedFiles is recreated so it's safe even if we switch threads between await.